### PR TITLE
SPV-1010/WebhookView

### PR DESCRIPTION
**Pull request by Nazarii-4chain**
# Pull Request Checklist

- [ ] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet-admin/blob/main/.github/CODE_STANDARDS.md)
- [ ] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet-admin/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [ ] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [ ] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -4,38 +4,38 @@
 
 "@4chain-ag/react-configuration@^1.0.4":
   version "1.0.4"
-  resolved "https://registry.yarnpkg.com/@4chain-ag/react-configuration/-/react-configuration-1.0.4.tgz#4fb9dc952fbd0af88d22731ff3a6adbbab2eec7f"
+  resolved "https://registry.npmjs.org/@4chain-ag/react-configuration/-/react-configuration-1.0.4.tgz"
```
Comment (chris-4chain) [inline yarn.lock:7]: Was it necessary to change from `yarnpkg` to `npmjs` ?

File: src/utils/webhooksQueryOptions.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+export interface WebhooksQueryOptions {
+  spvWalletClient: SpvWalletClientExtended;
+}
+
+export const webhooksQueryOptions = (opts: WebhooksQueryOptions) => {
+  const { spvWalletClient } = opts;
+
+  return queryOptions({
+    queryKey: ['webhooks', opts],
+    queryFn: async () => await spvWalletClient.AdminGetWebhooks(),
```
Comment (chris-4chain) [inline src/utils/webhooksQueryOptions.tsx:13]: `spvWalletClient.AdminGetWebhooks()` already returns a promise, the `async/await` pattern is unnecessary unless you plan to manipulate the result inside the function.
```suggestion
    queryFn: () => spvWalletClient.AdminGetWebhooks(),
```

File: src/components/ViewDialog/ViewDialog.tsx [hunk 3]
```diff
@@ -30,10 +30,10 @@ export const ViewDialog = ({ row }: ViewDialogProps) => {
           </div>
         );
       }
-
       return (
         <div key={field} className="break-all">
-          <span className="text-gray-400">{field}:</span> {value as React.ReactNode}
+          <span className="text-gray-400">{field}:</span>{' '}
+          {typeof value === 'boolean' ? value.toString() : (value as React.ReactNode)}
```
Comment (dzolt-4chain) [inline src/components/ViewDialog/ViewDialog.tsx:36]: Not a real issue but just curious. Why do we need to cast it here as ReactNode?

File: src/routes/admin/_admin.webhooks.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import { useState } from 'react';
+
+export const Route = createFileRoute('/admin/_admin/webhooks')({
+  component: Webhooks,
+  errorComponent: ({ error }) => <CustomErrorComponent error={error} />,
+});
+
+export function Webhooks() {
+  const { spvWalletClient } = useSpvWalletClient();
+  const [tab, setTab] = useState<string>('all');
```
Comment (dzolt-4chain) [inline src/routes/admin/_admin.webhooks.tsx:24]: I think here we don't need to type the string. Ts will infer the type. Please correct me if I'm wrong

File: yarn.lock (hunk 1 above)
Comment (Nazarii-4chain) [reply yarn.lock:7]: Reverted

File: src/utils/webhooksQueryOptions.tsx (hunk 2 above)
Comment (Nazarii-4chain) [reply src/utils/webhooksQueryOptions.tsx:13]: I've used the same approach in other queries. Also it helps to debug. However in this case let's do suggested change.

File: src/components/ViewDialog/ViewDialog.tsx (hunk 3 above)
Comment (Nazarii-4chain) [reply src/components/ViewDialog/ViewDialog.tsx:36]: without `ReactNode` it throws an error
`TS2322: Type unknown is not assignable to type ReactNode`

File: src/routes/admin/_admin.webhooks.tsx (hunk 4 above)
Comment (Nazarii-4chain) [reply src/routes/admin/_admin.webhooks.tsx:24]: Yes. Removed string type

**Review by chris-4chain — APPROVED** [review]

**Review by dzolt — APPROVED** [review]

File: src/components/UnsubscribeWebhook/index.ts
Comment (chris-4chain) [inline src/components/UnsubscribeWebhook/index.ts:1]: **Manual tests result**
Actions button and "View" modal is not necessary 
- All the informations are already in the table.
image

`a webhook` is moved to a new line
image

**Review by chris-4chain — CHANGES_REQUESTED** [review]

File: src/components/UnsubscribeWebhook/index.ts
Comment (Nazarii-4chain) [reply src/components/UnsubscribeWebhook/index.ts:1]: Updated

**Review by chris-4chain — APPROVED** [review]

**Pull request merged**